### ISSUE TITLE
NumericEdit display initial value limited by Decimals count.

### ISSUE THREAD

**Issue by WayneHiller** [stsrki/Blazorise#1841]
I have a NumericEdit like this:
```
<NumericEdit TValue="@decimal"
     Value="@detail.Price2"
     Decimals="2"
     ChangeTextOnKeyPress="true"
     ValueChanged="@(p => Price2Changed(detail, p))"/>

```

Even though it specifies Decimals="2" (the default) the initial value is displayed as 0.0000 or 99.9900

**Comment by stsrki** [comment]
I cannot reproduce it: Maybe it's because you have a different culture in your system. Try changing it:

```diff
<NumericEdit TValue="@decimal"
     Value="@detail.Price2"
     Decimals="2"
     ChangeTextOnKeyPress="true"
+    Culture="en-US"
     ValueChanged="@(p => Price2Changed(detail, p))"/>
```

**Comment by WayneHiller** [comment]
Adding the Culture did not change anything.

Ok this is interesting. These price values are retrieved from a database where they are stored as numeric(19, 4). When fetched via an API they display as 769.88 in the browser dev tools but display as 769.8800 in the NumericEdit. Changing the API to return Math.Round(Price1, 2) correctly displays 769.88 in the NumericEdit.

**Comment by stsrki** [comment]
Ahhh so it probably directly binds the value and just skips the internal validator that happens only on keypress. In that case, I should probably reformat it in `OnInitialize` also.

**Comment by WayneHiller** [comment]
It is interesting that value?.ToString() returns 769.8800 with the zeros, I guess it is just displaying the number of decimal places.

**Comment by stsrki** [comment]
I'm still not sure if I should commit to this. Or is it an issue in the first place. Do we really need to modify binded original value on purpose?

What I mean is. If we have `769.8877` in the DB and we bind it to a NumericEdit and if we do nothing with the input field we are good to save that same number back into the DB.

But if we change it on bind, and truncate it to `769.88` then we have data loss.

Is this the right way to do this? What if didn't want this to happen?

**Comment by WayneHiller** [comment]
I think it is the right thing to do if the Decimals value is set. So if I wanted to keep 4 decimals I would just set Decimals = 4. The way it is now (Decimals = 2) It will stay as 99.9900 until I edit the value. It seems more logical if Decimals was nullable and not initially set but that would probably be too much of a breaking change.

**Comment by sks-ttaylor** [comment]
@stsrki I also finding that Culture="en" or Culture="en-US" does not format the data

**Comment by stsrki** [comment]
Closing since I already fixed it with #2176
